### ISSUE TITLE
'incorrect number of dimensions' in scrna_evalPlot_DR

### ISSUE THREAD

**Issue by ansonrel** [plger/pipeComp#4]
Hi PL, 

Got this message when trying the readme: 

```{r}
scrna_evalPlot_DR(res, scale=FALSE)
```
```
Error in res[, grep("stepElapsed", colnames(res), invert = TRUE)] : 
  incorrect number of dimensions
4: as.matrix(res[, grep("stepElapsed", colnames(res), invert = TRUE)])
3: .prepRes(res, agg.by, agg.fn, elapsed = what == "elapsed")
2: scrna_evalPlot_DR(res, "silhouette", scale = FALSE, reorder_rows = reorder_rows, 
       value_format = value_format, show_heatmap_legend = TRUE, 
       col_title_fontsize = col_title_fontsize, agg.by = agg.by, 
       agg.fn = agg.fn, ...)
1: scrna_evalPlot_DR(res, scale = FALSE)
```

Coming from the `.prepRes` function that is called line 37. 

Going in debug mode, I **think** that the error comes from the list check line 35: 

```{r}
if (is.list(res)) 
    res <- res[[1]]
```

At this point, `res` is a data.frame and I guess you want to target "true" lists at this point. However `is.list` also captures data.frames. 

If it's true, then I guess that a `is(class(res) == "data.frame")`may do the trick. I tried it and it seems to resolve the issue with `.prepRes`.

**Comment by ansonrel** [comment]
Hi, coming back to the problem; 
I checked a bit further the code and one problem I see is indeed the rows of the heatmap. For the `silhouette` and `covarRes` options, the rownames correspond to the aggregated methods (eg, `"doubletmethod=none;filt=filt.lenient;norm=norm.seurat;[...]" `, which give the 3 methods as seen on the example figure of the readme. However when `switch` is called on `what="varExpl"`, it returns the `PCtop5.R2` of the top 30 PCs for the 3 datasets. 

I suspect either that 1) the formatting of this `PCtop5.R2` has been changed and we need to change the code of `scrna_evalPlot_DR` accordingly or 2) that the variance explained is in another slot of the result object. Maybe in the `clustering` slot ? It has different values such as "MI", "NMI", "NVI", but I don't know if the explained variance is in there ...
